### ISSUE TITLE
No setup option for users who have already configured a message

### ISSUE THREAD

**Issue by bethanylang** [google/site-kit-wp#7289]
## Feature Description

As reported by @wpdarren in Bug Bash:

We could simplify the set up for the case where a user already has set up an ABR message in AdSense by providing a secondary option to skip this part, similar to the post-click version of this today (2b).

2a) Initial state 

![image](https://github.com/google/site-kit-wp/assets/12189958/f0b7cbbe-a5db-4d35-b6e1-68c09ba7a646)

2b) Post-click

![image](https://github.com/google/site-kit-wp/assets/12189958/6f25e821-cc25-40bf-bdc5-646713dccd52)

That is, we could change 2a) to be the reverse of 2b), like so

![image](https://github.com/google/site-kit-wp/assets/12189958/5c681084-2431-4ad6-8f10-700de8c1e237)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The Step 2 of of Ad Blocking Recovery setup ([before the primary CTA is clicked](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-adsense-components-setup-adblockingrecoveryapp--step-two)) should have a secondary CTA with the label "I published my message" (note this differs from the mockup above which shows the label as "My message is ready").
    * The CTA should behave the same as the primary CTA ("My message is ready") of step 2 ([after the primary CTA is clicked](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-adsense-components-setup-adblockingrecoveryapp--step-two-after-cta-click)) - that is, set the Ad Blocking Recovery setup status to `setup-confirmed` and navigate user to the Dashboard with the success notification.
    * The newly added CTA should track an event using the existing `confirm_message_ready` event name but with an additional label to distinguish it from the Primary version of this event. ie. `secondary_cta`.

## Implementation Brief

In `assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/steps/CreateMessageStep.js`:

* Create a new `onSecondaryCTAClick` callback function with the following:
    * Copy all the logic except the first `if` condition block from the `onCTAClick` callback function.
    * Replace the `confirm_message_ready` event label with `confirm_message_ready_secondary_cta` in the `trackEvent` call.
* Locate the else condition of the `createMessageCTAClicked` ternary operator.
* Render a new `Link` component with the copy "I published my message".
* Pass the `onSecondaryCTAClick` callback to the `onClick` prop of the `Link` component.

### Test Coverage

* In `assets/js/modules/adsense/components/setup/AdBlockingRecoveryApp/SetupMain.test.js`, add tests for the `I published my message` button similar to the existing tests for the `My message is ready` button.
* Update VRT images.

## QA Brief

* Setup Site Kit and AdSense module
* Enter the setup for Ad Blocking recovery. It can be also accessed directly via url (`wp-admin/admin.php?page=googlesitekit-ad-blocking-recovery`)
* Verify that on the step 2, there is a new button `I published my message` next to the `create message` CTA

## Changelog entry

* Update the Ad Blocking Recovery setup to let users confirm setup if they have already done it.

**Comment by techanvil** [comment]
Hi @kuasha420, thanks for drafting the AC here.

It's a small point, but I would suggest that `create_message` and `confirm_message` are quite implementation-specific terms and are a little out of place in the AC. It _is_ pretty clear what they refer to, but it's preferable to use language that's more descriptive in terms of the flow rather than referring to these underlying implementation details.

Secondly, the final point is a little unclear in its wording and could with some clarification (I think it's just a case of updating it to read "track the event" instead of "share"?). Also, as the intent here does appear to be to track this event with a new label, we should keep @marrrmarrr in the loop here with a view to confirming the event variant and keeping track of it in the events [sheet](https://docs.google.com/spreadsheets/d/1eEuJ2ppVZuai1_hL77RA2vL7DUidM_qCnvTsjxZ_S4k/edit#gid=925918307&range=A142).

**Comment by kuasha420** [comment]
@techanvil Thank you for your feedback. 

1. You're right about the `create_message` and `confirm_message` are slightly implementation specific - but as both of them are part of step 2, saying step 2 before clicking and after clicking the primary CTA sounds more vague than this to me, and I think this make sense here. 

2. I've updated the wording here, and will let Mariya know about this so she can update the Event tracking sheet once the AC is approved.

Cheers.

**Comment by techanvil** [comment]
Hey @kuasha420, thanks for following up here.

> 1. You're right about the `create_message` and `confirm_message` are slightly implementation specific - but as both of them are part of step 2, saying step 2 before clicking and after clicking the primary CTA sounds more vague than this to me, and I think this make sense here. 

I do take your point, but thinking about reading the AC from another perspective, say from a QA point of view, I'm still not sure it's all that clear. Maybe another way of describing it would be to talk about Step 2 before and after the CTA click, and also link in to Storybook to [illustrate](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-adsense-components-setup-adblockingrecoveryapp--step-two) the [point](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-adsense-components-setup-adblockingrecoveryapp--step-two-after-cta-click). What do you think?

> 2. I've updated the wording here, and will let Mariya know about this so she can update the Event tracking sheet once the AC is approved.

Thanks for updating the wording and letting Mariya know about the event. Sorry if this sounds pedantic, though, but I would still suggest the wording could be a _little_ clearer - just updating it to start with "The newly added CTA should track an event using the existing `confirm_message_ready` event name" would help to make it very clear its supposed to track a new event.

**Comment by techanvil** [comment]
Thanks @kuasha420, this LGTM - I have made one small further adjustment to make it explicit that that new CTA should have the same text as the primary CTA, i.e. "My message is ready".

AC :white_check_mark:

**Comment by kuasha420** [comment]
@techanvil Thank you, I'm a bit confused by the "that new CTA should have the same text as the primary CTA", as far as I remember, even though that's the case in the mock screenshot, we decided to use the more explicit wording as described in the AC, as discussed in one of our AC sync. Maybe @aaemnnosttv would be able to confirm.

Cheers.

**Comment by techanvil** [comment]
Ah, my apologies @kuasha420. I had intended to save another back and forth but should have sent it back to you rather than making this change myself.

I had thought based on a previous discussion about CTAs that we were aiming to avoid relatively long text content within them. But, if we are indeed using the text "I have already created and published a message in AdSense" as the CTA, I think the AC could be a touch more explicit to make this entirely clear, what with the mockup showing a different label for the button.

I've tweaked it myself, please see what you think. Maybe we should then ask Evan to do the final ACR based on your comment above.

**Comment by kuasha420** [comment]
Thank you @techanvil. AC LGTM. Assigning this over to Evan for final pass. Cheers.

**Comment by techanvil** [comment]
@kuasha420, does that answer your question or do you want to check with Mariya on this one?

**Comment by tofumatt** [comment]
ACs here sound good 👍🏻

**Comment by derweili** [comment]
We have another issues here https://github.com/google/site-kit-wp/issues/7338 that refactors the same component. Unsure how to to write the IB here knowing that the relevant function will be changed by this other issue.

@bethanylang does it make sense to either set a dependency here or combine the two tasks?

**Comment by bethanylang** [comment]
Thanks for flagging, @derweili! @kuasha420 Can you please take a look at this versus #7338 and determine next steps here? Assigning this over to you for now.

**Comment by kuasha420** [comment]
@derweili Thanks for flagging, the flagged issues are fairly intertwined, and since the other one already seem to have a partial IB, I'm making this one depend on #7338 to make things easier here. Cheers.

**Comment by techanvil** [comment]
Thanks @hussain-t, this IB looks :white_check_mark:, but before moving to the EB I've asked @marrrmarrr on [Slack](https://10up.slack.com/archives/CFFRMC5DE/p1702981017454839) to confirm the new GA event we've got specced here.

**Comment by techanvil** [comment]
The GA event is confirmed, I'm moving this to EB :+1:

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified that "I published my message"  CTA added successfully to second step of ABR setup.
- Verified that  "I published my message"  CTA behaviour and design is as per AC.

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/e17b8632-2c2b-44b6-8012-9b2cfe106556)


![image](https://github.com/google/site-kit-wp/assets/94359491/5d56c327-7e3b-46b6-9ac3-9fda29a5b9d6)


https://github.com/google/site-kit-wp/assets/94359491/e3857366-dbe7-4de3-aeeb-f7dd48dcdfb6


https://github.com/google/site-kit-wp/assets/94359491/07cfd490-9703-4c0b-bc05-a813d8b36bda

</details>
